### PR TITLE
configuration: Renamed "Arg" to "Plugin"

### LINKED ISSUE #326: configuration: Change "arg" to "plugin"
`dffml/base.py` and some other places interpret `dict`s with `"arg"` and `"config"` keys in a special way.

`"arg"` is the name of the plugin to load, and `"config"` is the config for that plugin.

Therefore, `"arg"` should really be `"plugin"`

https://github.com/intel/dffml/blob/01d921ce610d809ef3a941d828c8d8c3b4b853ba/dffml/util/data.py#L21-L49

### PR DESCRIPTION
**Pull request by sakshamarora1**
Fixes #326

### CODE REVIEW

File: scripts/docs.py [hunk 1]
```diff
@@ -72,20 +72,24 @@ def sanitize_default(default):
 def build_args(config):
     args = []
     for key, value in config.items():
-        arg = value["arg"]
-        if arg is None:
+        plugin = value["plugin"]
```
Comment (pdxjohnny) [inline scripts/docs.py:75]: This is correct, nice job

File: service/http/docs/api.rst [hunk 2]
```diff
@@ -237,16 +237,16 @@ The following is an example request body to configure a model. The URL this
 
   {
     "model": {
-      "arg": null,
+      "plugin": null,
       "config": {
         "directory": {
-          "arg": [
+          "plugin": [
```
Comment (pdxjohnny) [inline service/http/docs/api.rst:243]: Also correct

File: tests/test_cli.py [hunk 3]
```diff
@@ -55,8 +55,8 @@ class RecordsTestCase(AsyncExitStackTestCase):
         self._stack.enter_context(
             patch.object(
                 ModelCMD,
-                "arg_model",
-                new=ModelCMD.arg_model.modify(type=model_load),
+                "plugin_model",
```
Comment (pdxjohnny) [inline tests/test_cli.py:58]:
```suggestion
                "arg_model",
```

File: tests/test_cli.py [hunk 4]
```diff
@@ -55,8 +55,8 @@ class RecordsTestCase(AsyncExitStackTestCase):
...
+                new=ModelCMD.plugin_model.modify(type=model_load),
```
Comment (pdxjohnny) [inline tests/test_cli.py:59]:
```suggestion
                new=ModelCMD.arg_model.modify(type=model_load),
```

File: source/mysql/dffml_source_mysql/source.py [hunk 5]
```diff
@@ -177,7 +177,7 @@ def args(cls, args, *above) -> Dict[str, Arg]:
             args,
             above,
             "record-query",
-            Arg(
+            Plugin(
```
Comment (pdxjohnny) [inline source/mysql/dffml_source_mysql/source.py:180]:
```suggestion
            Arg(
```

File: dffml/cli/cli.py [hunk 6]
```diff
@@ -50,10 +50,10 @@ class Merge(CMD):
     Merge record data between sources
     """
 
-    arg_dest = Arg(
+    plugin_dest = Plugin(
         "dest", help="Sources merge records into", type=BaseSource.load_labeled
     )
-    arg_src = Arg(
```
[the remark this answers is not in the document]
Comment (sakshamarora1) [reply dffml/cli/cli.py:56]: Ohh! My bad!! Sorry, I'll correct them

**Comment by pdxjohnny** [comment]
@sakshamarora1 can you merge master and put this under the unreleased section?

**Pull request merged**